strawberryfields/gbs/vibronic.py

```diff
@@ -0,0 +1,228 @@
+corresponding vacuum states by applying creation operators. Starting from the ground state of a
+molecule at 0 K, which corresponds to the vacuum of ground state vibrational excitations
+:math:`| \mathbf{0} \rangle`, the Franck-Condon factors can be written as:
+
+.. math::
+    FCF = \left | \left \langle \mathbf{m} |U_{Dok}| \mathbf{0} \right \rangle \right | ^ 2,
+
+where the sate :math:`| \mathbf{m} \rangle = | m_1,...m_k \rangle` contains information to obtain
+the transition energy that corresponds to the FCF through counting the number of vibrational quanta
+:math:`m` in :math:`k` modes. This is equivalent to the probability of observing output patterns
+of a GBS apparatus constructed according to :math:`U_{Dok}`. The Doktorov operator contains
+parameters that should be obtained from quantum chemistry calculations for a given molecule.
+Designing a GBS device with such molecular parameters allows obtaining the Franck-Condon profile
+of the molecule which can be used to construct the molecular vibronic spectra.
+
+The Doktorov operator can be written in terms of displacement (:math:`D_\alpha`), squeezing
+(:math:`\Sigma`) and interferometer (:math:`C_L, C_R^t`) operators as
+
+.. math::
+    U_{Dok} = C_L \Sigma C_R^t D_\alpha.
+
+The squeezing and rotation operators are obtained by singular value decomposition of a matrix
+:math:`J` as
+
+.. math::
+    J = C_L \Sigma C_R^t.
+
+In order to obtain :math:`J` we need to construct diagonal matrices :math:`\Omega` and
+:math:`\Omega'` from square roots of the ground state and excited state normal mode frequencies,
+:math:`\omega` and :math:`\omega'`, respectively. Then :math:`J` is obtained as
+
+.. math::
+    J = \Omega' U \Omega^{-1},
+
+where
+
+.. math::
+    \Omega = diag (\sqrt{\omega_1},...,\sqrt{\omega_k})
+
+    \Omega' = diag (\sqrt{\omega_1'},...,\sqrt{\omega_k'})
+
+and :math:`U` is a molecular coordinate transformation known as the Duschinsky matrix which is
+obtained from the normal mode coordinates of the ground and excited states, :math:`q` and
+:math:`q'` respectively, as
+
+.. math::
+    q' = Uq + d.
+
+The displacement parameter :math:`d` is related to the structural changes of a molecule upon
+vibronic excitation and is related to the displacement parameter :math:`\alpha` as
+
+.. math::
+    \alpha = d / \sqrt{2}.
+
+In general, all the GBS parameters can be derived from the previously computed equilibrium
+geometries, vibrational frequencies and normal coordinates of the molecule in its (electronic)
+ground and excited states.
+
+
+Summary
+-------
+
+.. autosummary::
+    gbs_params
+    energy
+    spectrum
+
+Code details
+^^^^^^^^^^^^
+"""
+from typing import Tuple
+
+import numpy as np
+
+
+def gbs_params(
+    w: np.ndarray, wp: np.ndarray, Ud: np.ndarray, d: np.ndarray
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    r"""Converts molecular information input to Gaussian gate parameters.
```

Comment:
```suggestion
    r"""Converts molecular information to GBS gate parameters.
```
This does leave out the face that we mean _the most general_ form of GBS, but I think it's ok here